### DISCUSSION TITLE
dict(字典)的鍵(keys)可以用什麼？

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#276] [General]
dict的keys，只要是使用值不可變的資料型態都可以。目前我知道的包含屬於文字有序型態(text sequence type)的字串(string, str)；屬於數值型態(numeric types)的整數(integer, int)、浮點數(floating numbers, float)與複數(complex numbers, complex)與屬於有序型態(sequence types)的元組(tuple)。

**Comment by chia-kai-chang** [comment]
沒有錯。

但是建議使用『字串String』作為dict的key值。

**Comment by enoch-lightning** [comment]
原生Python中值不可變的資料型態
- 文字有序型態(text sequence type)

1. 字串，英文單數為string，複數為strings。使用Python的type()函式，看到str代表string。
- 數值型態(numeric types)

1. 整數，英文單數為integer，複數為integers。使用Python的type()函式，看到int代表integer。
2. 浮點數，英文單數為floating number，複數為floating numbers。使用Python的type()函式，看到float代表floating number。
3. 複數，英文單數為complex number，複數為complex numbers。使用Python的type()函式，看到complex代表complex number。
- 有序型態(sequence types)

1. 元組，英文單數為tuple，複數為tuples。使用Python的type()函式，看到tuple代表tuple。

**Comment by enoch-lightning** [comment]
這邊結合函式與字典，製作一個範例給大家看。
![image](https://user-images.githubusercontent.com/86505454/148358746-02758a84-7c6c-49c4-af78-814d6612340f.png)
